checkstyle-tester/diff.groovy

```diff
@@ -0,0 +1,262 @@
+import java.nio.file.Paths
+
+import static java.lang.System.err
+
+static void main(String[] args) {
+    def cli = new CliBuilder(usage:'groovy diff.groovy [options]')
+    cli.with {
+        r(longOpt: 'localGitRepo', args: 1, argName: 'localGitRepo', 'Path to local git repository')
+        b(longOpt: 'baseBranch', args: 1, argName: 'baseBranch', 'Base branch name. Default is master')
+        p(longOpt: 'patchBranch', args: 1, argName: 'patchBranch', 'Name of the patch branch in local git repository')
+        c(longOpt: 'checkstyleCfg', args: 1, argName: 'checkstyleCfg', 'Path to checkstyle config file')
+        l(longOpt: 'projectsToTestOn', args: 1, argName: 'projectsToTestOn', 'Path to file which contains projects to test on')
+    }
+    def options = cli.parse(args)
+
+    if (areValidCliArgs(options)) {
+        def localGitRepo = new File(options.localGitRepo)
+        if (hasUnstagedChanges(localGitRepo)) {
+            throw new IllegalStateException("Error: git repository ${localGitRepo.getPath()} has unstaged changes!")
+        }
+        def testerCheckstyleVersion = getCheckstyleVersionFromPomXml('./pom.xml', 'checkstyle.version')
+        def checkstyleVersionInLocalRepo = getCheckstyleVersionFromPomXml("$localGitRepo/pom.xml", 'version')
+        if (testerCheckstyleVersion != checkstyleVersionInLocalRepo) {
```

Comment:
The full point is not addressed.
You are checking pom in checkstyle git, but not for master and patch branches. You are checking it for some random initial branch. Since no checkouts have been called, we can't tell what we are running this check on.

https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/launch_diff.sh#L118-L129
`launch_diff` checks the POM for both branches. It is in `launch` function which is called twice.
Master: https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/launch_diff.sh#L363
Patch: https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/launch_diff.sh#L398